### PR TITLE
Another set of general fixes

### PR DESCRIPTION
**Pull request by tsurdilo**
Another set of general fixes (typos, formatting, rewording).
Only notable thing maybe is I moved the "use cases" section to the bottom of the document.
This is because currently use cases section needs an overhaul and is too complex to have right at the top of the page. Once revised we should move it back to the top.

**added** - fixed loop parameter in documentation to not be required, as per discussion in https://github.com/cncf/wg-serverless/issues/140

### CODE REVIEW

**Comment by tsurdilo** [comment]
rebased against master

File: workflow/spec/spec.md [hunk 1]
```diff
@@ -327,8 +321,8 @@ see the [Workflow Error Handling section](#Workflow-Error-Handling).
 | Parameter | Description | Type | Required |
 | --- | --- | --- | --- |
 | [condition](#Condition-Definition) | Condition that matches against the error in the state data input | string |yes |
-| [filter](#Filter-Definition) |Error data filter | object | yes |
-| [transition](#Transitions) |Next transition of the workflow when an errors expressed in errorExpression are matched | string | yes |
+| [filter](#Filter-Definition) | Error data filter | object | yes |
+| [transition](#Transitions) |   when an errors expressed in errorExpression are matched | string | yes |
```
Comment (cathyhongzhang) [inline workflow/spec/spec.md]: I think this might be a typo. Needs to add back "Next transition of the workflow ..."

File: workflow/spec/spec.md [hunk 2]
```diff
@@ -389,11 +421,11 @@ We will start defining each individual state:
...
 | type | State type | string | yes |
-| end |Is this state an end state | boolean | no |
-| [events](#eventstate-eventdef) |State events | array | yes |
-| [filter](#Filter-Definition) |State data filter | object | yes |
-| [loop](#Loop-Definition) |State loop information | object | yes |
-| [onError](#Workflow-Error-Handling) |States error handling definitions | array | no |
+| end | Is this state an end state | boolean | no |
+| [events](#eventstate-eventdef) | State events | array | yes |
+| [filter](#Filter-Definition) | State data filter | object | yes |
+| [loop](#Loop-Definition) | State loop information | object | no |
```
Comment (cathyhongzhang) [inline workflow/spec/spec.md]: You are going to move this loop out of the state, right?

File: workflow/spec/spec.md [hunk 3]
```diff
@@ -1303,29 +1293,30 @@ Let's define a branch now:
...
 </details>
 
+
 Each branch receives a copy of the Parallel state's input data.
-Transitions for states within a branch can only be to other states in that branch. 
-In addition, states outside a Parallel state cannot transition to a state within a branch of a Parallel state.
-The Parallel state generates an output array in which each element is the output for a branch. 
-The elements of the output array need not be of the same type.
+States within each branch are only allowed to transition to states defined in the same branch. 
+Transitions to other branches or workflow states are not allowed.
```
Comment (cathyhongzhang) [inline workflow/spec/spec.md]: missing "states outside a Parallel state cannot transition to a state within a branch of a Parallel state"

File: workflow/spec/spec.md [hunk 4]
```diff
@@ -1812,35 +1804,37 @@ The default should be that if this happens workflow execution should halt and a
 
 ## Workflow Data
 
-Workflow data is represented in [JSON](https://www.json.org/json-en.html) format. Flow of data during
-serverless workflow execution can be divided into the following constructs:
+Serverless Workflow data is represented in [JSON](https://www.json.org/json-en.html) format. 
+Flow of data during workflow execution can divided into:
```
Comment (cathyhongzhang) [inline workflow/spec/spec.md]: change to -->be divided

File: workflow/spec/spec.md (hunk 1 above)
Comment (tsurdilo) [reply workflow/spec/spec.md]: fixed

File: workflow/spec/spec.md (hunk 2 above)
Comment (tsurdilo) [reply workflow/spec/spec.md]: yeah that will be part of another pr after this one is merged

File: workflow/spec/spec.md (hunk 3 above)
Comment (tsurdilo) [reply workflow/spec/spec.md]: added

File: workflow/spec/spec.md (hunk 4 above)
Comment (tsurdilo) [reply workflow/spec/spec.md]: fixed

**Comment by tsurdilo** [comment]
@cathyhongzhang updated

**Comment by cathyhongzhang** [comment]
LGTM

**Pull request merged**